openshift-201/rh201-pod-auto-scale.md

```diff
@@ -0,0 +1,283 @@
+
+The VPA CR must be in the same project as the pods you want to monitor. There are 4 updateModes available for the VPA:
+
+* Auto, the VPA assigns resource requests on pod creation and updates the existing pods by terminating them when the requested resources differ significantly from the new recommendation.
+* Recreate, the VPA assigns resource requests on pod creation and updates the existing pods by terminating them when the requested resources differ significantly from the new recommendation. This mode should be used rarely, only if you need to ensure that the pods are restarted whenever the resource request changes. Otherwise prefer the "Auto" mode which may take advantage of restart free updates once they are available.
+* Initial, automatically applies VPA recommendations only at pod creation.
+* Off, only provides recommended resource limits and requests, allowing you to manually apply the recommendations. The off mode does not update pods.
+
+
+Lets first confirm that the VPA operator is installed in our cluster, we should see 2 results.
+
+```
+oc api-resources | grep vpa
+```
+### Update Mode Off
+
+Lets now create a VPA with update mode turned to "Off" so we just get resource recommendations.
+
+Create a VPA using the below yaml:
+
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: vpa-recommender
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind:       Deployment 
+    name:       hello-world-nginx
+  updatePolicy:
+    updateMode: "Off"
+```
+
+Re-trigger load-test deployment by updating the environment variable REQUESTS to a different number. Let the deployment fire up a new pod and send load to our hello-world-nginx pod.
+
+Give it a few minutes and check the vpa status sections for recommendations.
+
+
+```yaml
+oc get vpa vpa-recommender -o yaml
+
+ status:
+    conditions:
+    - lastTransitionTime: "2022-04-11T17:35:54Z"
+      status: "True"
+      type: RecommendationProvided
+    recommendation:
+      containerRecommendations:
+      - containerName: hello-world-nginx
+        lowerBound:
+          cpu: 25m
+          memory: 262144k
+        target:
+          cpu: 35m
+          memory: 262144k
+        uncappedTarget:
+          cpu: 35m
+          memory: 262144k
+        upperBound:
+          cpu: 66m
+          memory: 262144k
+```
+
+
+The output shows the recommended resources under target.
+The minimum recommended resources under lowerBound. 
+The highest recommended resources under upperBound.
+The most recent resource recommendations under uncappedTarget.
```

Comment:
Markdown will eat these newlines and turn it into one line. Suggest turning into a list `-`

Reply:
Done. Thank you!